### PR TITLE
Clarify `astro check` re: type checking

### PR DESCRIPTION
**Pull request by happycollision**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [x] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [ ] Something else!

#### Description

The old messaging in the documentation for `astro check` indicates that it does type checking. It does not:

https://github.com/withastro/docs/blob/d96e48c31b92a6be3c4ad632111f6d0d4764209b/src/pages/en/guides/typescript.md?plain=1#L10

So this clarifies that fact, and links back to the page explaining how to integrate TypeScript.

### CODE REVIEW

**Comment by happycollision** [comment]
Is `astro check` being removed? That merged commit suggests it is. If so, you can close this PR.

**Comment by sarah11918** [comment]
No, but I did just need to resolve a conflict because this page had changed... let me make sure it didn't get lost! Thanks for the heads up!

**Comment by sarah11918** [comment]
@happycollision I think the original `astro-check`, and your propsed edits, are still there? I'm still seeing this in the Files Changed for this PR, which I think is what we expect.

I did just (separately) merge a PR that added two new commands below `astro-check` (-add, and -docs). Let me know if anything doesn't look right on your end. 

Screenshot 2022-05-18 12 28 37

**Comment by happycollision** [comment]
Whoops! Yes. I thought I was viewing the whole branch's diff, not just the merged one. Confused myself.

**Comment by sarah11918** [comment]
BEEN THERE!  😄 

But, I'm checking on this behaviour now, and I'm being told that in fact this command *should* type check:

> What this command does is just show the diagnostics you would see the editor in the console. It uses an old version of the language server that didn't support script tags, maybe that could be what is confusing? It also only checks astro files.

So, with that new information on how it *should* work, is there something else you're expecting from this description that you're not getting to work? Is it that it doesn't type check for React etc. files, only Astro files? Or, is it possible that maybe you've encountered a bug with `astro-check` itself, and not something that it doesn't do?  

Just trying to figure out the best thing to have in the docs, because it seems like saying it does type checking *is* what's intended. Thanks for your help with this!

**Comment by happycollision** [comment]
Here's a visual comparison, since I added a blockquote in the latest iteration.

|before|after|
|-|-|
|CleanShot 2022-05-18 at 22 33 42@2x|CleanShot 2022-05-18 at 22 32 43@2x|

I'll squash and rebase this onto one commit if all looks good.

File: src/pages/en/reference/cli-reference.md [hunk 1]
```diff
@@ -33,10 +33,13 @@ This command is meant for local testing only, and is not designed to be run in p
 
 ### `astro check`
 
-Runs diagnostics (such as type-checking) against your project and reports errors to the console. If any errors are found the process will exit with a code of **1**.
+Runs diagnostics (such as type-checking within .astro files) against your project and reports errors to the console. If any errors are found the process will exit with a code of **1**.
```
Comment (sarah11918) [inline src/pages/en/reference/cli-reference.md:36]:
```suggestion
Runs diagnostics (such as type-checking within `.astro` files) against your project and reports errors to the console. If any errors are found the process will exit with a code of **1**.
```

File: src/pages/en/reference/cli-reference.md [hunk 2]
```diff
@@ -33,10 +33,13 @@ This command is meant for local testing only, and is not designed to be run in p
...
 
 This command is intended to be used in CI workflows.
 
+> Note that this command only checks types within .astro files.
```
Comment (sarah11918) [inline src/pages/en/reference/cli-reference.md:40]:
```suggestion
> Note that this command only checks types within `.astro` files.  
```

File: src/pages/en/reference/cli-reference.md [hunk 3]
```diff
@@ -33,10 +33,13 @@ This command is meant for local testing only, and is not designed to be run in p
 
 ### `astro check`
 
-Runs diagnostics (such as type-checking) against your project and reports errors to the console. If any errors are found the process will exit with a code of **1**.
+Runs diagnostics (such as type-checking within .astro files) against your project and reports errors to the console. If any errors are found the process will exit with a code of **1**.
 
 This command is intended to be used in CI workflows.
 
+> Note that this command only checks types within .astro files.  
+> 📚 Read more about [TypeScript support in Astro.](/en/guides/typescript/)
```
Comment (sarah11918) [inline src/pages/en/reference/cli-reference.md:41]:
```suggestion
> 📚 Read more about [TypeScript support in Astro](/en/guides/typescript/).
```

**Comment by sarah11918** [comment]
Hey, thanks for this @happycollision ! And, I like that you used our pattern of emoji + call out to read more about something.

I just made a couple of very tiny style corrections as suggestions to match the conventions of the rest of the site, and then I think this is good to go.

**Comment by sarah11918** [comment]
Thanks for being a contributor to the docs, @happycollision ! You should soon see yourself on the main page in our FacePile!  :)

**Pull request merged**